### ISSUE TITLE
SyntaxError: Unexpected end of input

### ISSUE THREAD

**Issue by recapp-developer** [MacroPower/macropower-analytics-panel#4]
Hi,

Thanks for this plugin, will be really useful for my work. I've configured it with my REST API backend that can receive the JSON body successfully. But after a few seconds, the plugin view reports a "SyntaxError: Unexpected end of input" error. My server returns a JSON response with the format {"location":"12345"} which seems to correspond with the guidance in the documentation.
<img width="869" alt="Screenshot 2020-07-14 at 11 59 03" src="https://user-images.githubusercontent.com/43670998/87418523-b8ef2480-c5c9-11ea-89db-41a09f6fde81.png">

I'm not sure if this is related but when I navigate away from the dashboard I never receive an 'end of session' response even though my server is set up to receive it (i.e. myaddress/mypath/{id}).

Did anyone experience this? I'm using Grafana 7.0.6 (9f4a172a76) and plugin version 1.0.0.

**Comment by MacroPower** [comment]
Hey there, so the error you're receiving is indicative of some issue parsing your JSON response. However, the format you've posted here looks correct. Perhaps your server replies with something like Content-Type 'text/html' rather than 'application/json'? If you could upload a HAR file that would probably tell the whole story.

I've also noticed an issue with the request I'm submitting - there should be an 'Accept: application/json' header, but there isn't. This might be the source of your issue depending on how your backend works. I can try to push a patch for this sometime this evening (EST).

**Comment by recapp-developer** [comment]
Hey, I've attached the HAR file and the response from my server which shows application/json is set as the content-type. Maybe something to do with the Transfer-Encoding ? I use the NiFi framework and I don't believe there's a way to configure it other than chunked.

The test API call I've setup can receive any content type at the moment.

Thx,
mike

**Comment by MacroPower** [comment]
That's interesting - upon first glance everything looks fine here. Transfer-Encoding might be an issue, but I'm not sure about that. Could you upload another har including all the relevant entries w/ data please? I'm only seeing the first call in your file and it doesn't include the response body. There might be an issue somewhere that I can't see.

**Comment by recapp-developer** [comment]
That is the full HAR output of the request when clicking "Retry" in the panel. Seems like it's not handling the response as when I inspect the request I'm unable to view the response body in web inspector. Says 'failed to load response data'. When testing my API call in Postman I get a valid response back which it parses ok.

**Comment by recapp-developer** [comment]
I also wondered whether it was an issue on the server hosting my Grafana platform connecting to my backend, so I replicated the POST request from it via curl and received the response as expected. My Grafana and REST API are on different servers, so I have noCors=true in the analytics panel.

**Comment by MacroPower** [comment]
Have you tried submitting a request with noCors=false? If so, what was the result of that?

**Comment by recapp-developer** [comment]
I get a resource error  "Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled."

**Comment by MacroPower** [comment]
Alright, I think I see what's going on. So, since you're interested in the response from the server, you _must_ have cors enabled. For more info see [this post](https://stackoverflow.com/questions/36292537/what-is-an-opaque-response-and-what-purpose-does-it-serve) about opaque responses. When I added the "No Cors" switch it was primarily for usage with Telegraf, since in that case an opaque response is just fine.

You're likely unable to fetch with cors enabled because (as the error suggests) your backend is not responding with a correct `Access-Control-Allow-Origin` header. Here's an example of what said response can look like:

Preflight:
![Grafana1](https://user-images.githubusercontent.com/5648814/87472508-0bdad300-c5ee-11ea-860e-3597de35a0c0.png)

Post:
![Grafana2](https://user-images.githubusercontent.com/5648814/87472926-ba7f1380-c5ee-11ea-9b25-0c17fa914c72.png)


In my case I set this up with the following config, but it will be different for you.

```
@Configuration
@EnableWebSecurity
class SecurityConfig : WebSecurityConfigurerAdapter() {
    @Throws(Exception::class)
    override fun configure(security: HttpSecurity) {
        security.httpBasic().disable()
        security.csrf().disable()
        security.cors().configurationSource { CorsConfiguration().applyPermitDefaultValues() }
    }
}
```
